src/coprocessor/Client.cc

```diff
@@ -115,7 +115,23 @@ std::vector<LocationKeyRanges> splitKeyRangesByLocations(
     return res;
 }
 
-std::vector<BatchCopTask> balanceBatchCopTasks(kv::RegionCachePtr & cache, std::vector<BatchCopTask> && original_tasks, bool is_mpp, const kv::LabelFilter & label_filter, Poco::Logger * log)
+std::map<uint64_t, kv::Store> filterAliveStores(kv::Cluster * cluster, const std::map<uint64_t, kv::Store> & stores, Logger * log)
+{
+    std::map<uint64_t, kv::Store> alive_stores;
+    for (const auto & ele : stores)
+    {
+        if (!cluster->mpp_prober->isRecovery(ele.second.addr, /*mpp_fail_ttl=*/std::chrono::seconds(60)))
+            continue;
+
+        if (!common::detectStore(cluster->rpc_client, ele.second.addr, /*rpc_timeout=*/2, log))
+            cluster->mpp_prober->add(ele.second.addr);
+
+        alive_stores.emplace(ele.first, ele.second);
```

Comment:
why detectStore return not available, still needs to add the store to `alive_stores`?